### PR TITLE
[CI Environment] Added controls to enable only Pester and or only deployment tests

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Description

- Added controls to both GitHub & ADO pipelines to enable a pipeline run with only Pester and or Deployment tests
- Updated an outdated screenshot
- Shortened GitHub deploy job name
- Added job names to ADO templates

# Examples:
## [Only Pester GitHub](https://github.com/Azure/ResourceModules/actions/runs/3969307154)
image

## [Only Pester Azure DevOps](https://dev.azure.com/carml/CARML/_build/results?buildId=19082&view=results)
image

## [Only Deploy GitHub](https://github.com/Azure/ResourceModules/actions/runs/3969487644)
image

## [Only Deploy Azure DevOps](https://dev.azure.com/carml/CARML/_build/results?buildId=19087&view=results)
image

| Pipeline |
| - |
| [AnalysisServices: Servers](https://github.com/Azure/ResourceModules/actions/workflows/ms.analysisservices.servers.yml) |
| [Build Status](https://dev.azure.com/carml/CARML/_build/latest?definitionId=2&branchName=users%2Falsehr%2F1372_runPar) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: .azuredevops/modulePipelines/ms.aad.domainservices.yml [hunk 1]
```diff
@@ -1,6 +1,14 @@
 name: 'AAD - DomainServices'
```
Comment (AlexanderSehr) [inline .azuredevops/modulePipelines/ms.aad.domainservices.yml:1]: Blocking comment until release 0.9

File: .github/workflows/ms.analysisservices.servers.yml [hunk 2]
```diff
@@ -69,6 +79,7 @@ jobs:
   job_module_pester_validation:
```
Comment (ahmadabdalla) [inline .github/workflows/ms.analysisservices.servers.yml:79]: @AlexanderSehr would it make sense to change the job name to match other names that use `staticValidation`. E.g ==> `job_module_staticValidation`  or `job_module_static_validation` ?

Comment (AlexanderSehr) [reply .github/workflows/ms.analysisservices.servers.yml:79]: Sure, why not. I guess the original reason was to imply that this runs Pester - but I don't think that's too important on that level. Will update

Comment (AlexanderSehr) [reply .github/workflows/ms.analysisservices.servers.yml:79]: Done. I went with `job_module_static_validation`

File: .github/workflows/ms.analysisservices.servers.yml [hunk 3]
```diff
@@ -123,6 +138,7 @@ jobs:
     if: github.ref == 'refs/heads/main' || github.ref == 'refs/heads/master' || github.event.inputs.prerelease == 'true'
```
Comment (eriqua) [inline .github/workflows/ms.analysisservices.servers.yml:138]: We shouldn't allow publishing unless both static and deployment validation run. Is that the case?

Comment (AlexanderSehr) [reply .github/workflows/ms.analysisservices.servers.yml:138]: It is. The job depends in the others without any special conditions that would change that

Comment (AlexanderSehr) [reply .github/workflows/ms.analysisservices.servers.yml:138]: Also retested it again running with the pre-release flag alongside only pester vs only deployment for both GitHub & ADO

**Review by ahmadabdalla — APPROVED** [review]

**Pull request merged**